scripts/generate-ebpf-types.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+# Copyright Antimetal, Inc. All rights reserved.
+#
+# Use of this source code is governed by a source available license that can be found in the
+# LICENSE file or at:
+# https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt
+
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+PROJECT_ROOT="$(cd "${SCRIPT_DIR}/.." && pwd)"
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[0;33m'
+NC='\033[0m' # No Color
+
+# Check if we're in the right directory
+if [[ ! -f "${PROJECT_ROOT}/go.mod" ]]; then
+    echo -e "${RED}Error: This script must be run from the system-agent repository root${NC}"
+    exit 1
+fi
+
+echo -e "${GREEN}Generating eBPF type definitions...${NC}"
+
+# Build the generator tool
+echo "Building ebpf-typegen..."
+if ! go build -o "${PROJECT_ROOT}/bin/ebpf-typegen" "${PROJECT_ROOT}/cmd/ebpf-typegen/main.go"; then
+    echo -e "${RED}Failed to build ebpf-typegen${NC}"
+    exit 1
+fi
```

Comment:
adding to the above comment, perhaps we should consider this an actual tool we can add to the makefile
```make
.PHONY: generate-ebpf-types
generate-ebpf-types: ebpf-typegen ## Generate Go types from eBPF header files
	@echo "Generating Go types from eBPF headers..."
	@$(ROOT)/scripts/generate-ebpf-types.sh

EBPF_TYPEGEN ?= $(LOCALBIN)/ebpf-typegen

.PHONY: ebpf-typegen
ebpf-typegen: $(EBPF_TYPEGEN) ## Build ebpf-typegen if neccessary.
$(EBPF_TYPEGEN):  FORCE # Always build and rely on go build cache
  go build -o "$(EBPF_TYPEGEN)" "$(ROOT)/tools/ebpf-typegen/main.go"
```